mitiq/vd/vd_utils.py

```diff
@@ -51,3 +55,137 @@ def map_for_grid_qubits(qu: cirq.Qid) -> cirq.Qid:
             new_circuit += circuit.transform_qubits(map_for_grid_qubits)
 
     return new_circuit
+
+
+def _apply_cyclic_system_permutation(
+    matrix: NDArray[np.complex128], N_qubits: int, M: int = 2
```

Comment:
```suggestion
    matrix: NDArray[np.complex64], N_qubits: int, M: int = 2
```

Lets use 64-bit precision throughout since we use that in other parts of the code.